src/domain/fiat/display-currency.ts

```diff
@@ -5,40 +5,44 @@ export const MajorExponent = {
   ZERO: 0,
   ONE: 1,
   THREE: 3,
+  FOUR: 4,
 } as const
 
-export const minorToMajorUnit = ({
+export const minorToMajorUnitFormatted = ({
   amount,
-  displayMajorExponent,
+  displayCurrency,
 }: {
   amount: number | bigint
-  displayMajorExponent: CurrencyMajorExponent
-}) => (Number(amount) / 10 ** displayMajorExponent).toFixed(displayMajorExponent)
-
-export const usdMinorToMajorUnit = (amount: number | bigint) =>
-  Number(minorToMajorUnit({ amount, displayMajorExponent: MajorExponent.STANDARD }))
+  displayCurrency: DisplayCurrency
+}): string => {
+  const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
+  const majorAmount = Number(amount) / 10 ** displayMajorExponent
+  return majorAmount.toFixed(displayMajorExponent)
+}
 
-export const majorToMinorUnit = ({
+export const minorToMajorUnit = ({
   amount,
-  displayMajorExponent,
+  displayCurrency,
+  fixed = true,
 }: {
   amount: number | bigint
-  displayMajorExponent: CurrencyMajorExponent
-}) => Number(Number(amount) * 10 ** displayMajorExponent)
-
-// TODO: remove after remove hardcoded DisplayCurrency.Usd
-export const usdMajorToMinorUnit = (amount: number | bigint) =>
-  majorToMinorUnit({ amount, displayMajorExponent: MajorExponent.STANDARD })
+  displayCurrency: DisplayCurrency
+  fixed?: boolean
+}): number => {
+  const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
+  const majorAmount = Number(amount) / 10 ** displayMajorExponent
+  return fixed ? Number(majorAmount.toFixed(displayMajorExponent)) : majorAmount
```

Comment:
I think the two values in this conditional work out to the same, redundant conditional maybe. Looks like the string stuff is covered with `minorToMajorUnitFormatted` anyway too.

Reply:
`fixed: false` it is needed for displayCurrencyPerSettlementCurrencyMajorUnit

Reply:
Oh right, the `fixed` name isn't intuitive though. ChatGPT recommended the following:
- [ ] useRounded
- [ ] applyPrecision
- [ ] roundResult
- [x] exactPrecision
- [ ] isRounded

I like the `exactPrecision` one